src/AppInstallerCommonCore/Public/AppInstallerLogging.h

```diff
@@ -171,6 +175,40 @@ namespace AppInstaller::Logging
     private:
         std::stringstream m_out;
     };
+
+    struct ThreadGlobals

```

Comment:
Should be in its own file; should not be in the Logging namespace.

Reply:
Agree that it should not be in logging namespace. Problem I found with having it in it's own file is it messes with header file inclusion order - Logging.h would need to include ThreadGlobals.h to refer to t_pThreadGlobals and ThreadGlobals.h would need to include Logging.h and Telemetry.h. So compiler complains about one of the two definitions not available.

Reply:
Not a problem when taken with the other suggestions:  Headers don't need to know about the global, because it isn't global.  Only the cpp file needs to know how to call `ThreadGlobals::GetCurrent` or similar.

Reply:
Gotcha